plume-web-jersey/src/main/java/com/coreoz/plume/jersey/security/control/ContentControlFeature.java

```
@@ -0,0 +1,162 @@
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.DynamicFeature;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ReaderInterceptor;
+import javax.ws.rs.ext.ReaderInterceptorContext;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class ContentControlFeature implements DynamicFeature {
+
+    private static final Logger logger = LoggerFactory.getLogger(ContentControlFeature.class);
+
+    @Override
+    public void configure(ResourceInfo resourceInfo, FeatureContext context) {
+        addContentSizeFilter(resourceInfo.getResourceMethod(), context);
+    }
+
+    private void addContentSizeFilter(AnnotatedElement annotatedElement, FeatureContext methodResourcecontext) {
+        ContentSizeLimit contentSizeLimit = annotatedElement.getAnnotation(ContentSizeLimit.class);
+        methodResourcecontext.register(new ContentSizeLimitInterceptor(
+                contentSizeLimit != null ? contentSizeLimit.value() : ContentSizeLimitInterceptor.DEFAULT_MAX_SIZE
+            ));
+    }
+
+    public static class ContentSizeLimitInterceptor implements ReaderInterceptor {
+
+        private static final int DEFAULT_MAX_SIZE = 500;
+
+        private final int maxSize;
+
+        public ContentSizeLimitInterceptor(int maxSize) {
+            this.maxSize = maxSize;
+        }
+
+        // https://stackoverflow.com/questions/24516444/best-way-to-make-jersey-2-x-refuse-requests-with-incorrect-content-length
+        @Override
+        public Object aroundReadFrom(ReaderInterceptorContext context) throws IOException {
+            final InputStream contextInputStream = context.getInputStream();
+            final String headerContentLength = context.getHeaders().getFirst("Content-Length");
+            final Long declaredContentLength = headerContentLength == null ? -1 : Long.valueOf(headerContentLength);
```

Ici on devrait avoir un système "fail fast" : si déjà le content length déclaré est supérieur à `maxSize`, on stoppe tout de suite la requête.

Et donc ensuite dans `ContentSizeLimitInputStream` en maxSize, on fournit `declaredContentLength` (sachant que s'il est supérieur à `maxSize`, on aura déjà levé une exception)